File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/presentation/ui/profile/help/HelpActivity.java
package com.lattis.ellipse.presentation.ui.profile.help;

import android.content.Intent;
import android.net.Uri;
import android.os.Build;
import androidx.annotation.NonNull;
import android.text.Html;

import com.lattis.ellipse.presentation.ui.base.activity.BaseBackArrowActivity;
import com.lattis.ellipse.presentation.view.CustomTextView;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.OnClick;
import io.lattis.ellipse.R;

public class HelpActivity  extends BaseBackArrowActivity<HelpPresenter> implements HelpView {
    private String phoneNumber;


    @BindView(R.id.tv_help_info)
    CustomTextView helpInfo;

    @Inject
    HelpPresenter helpPresenter;
    @Override
    protected void inject() {
        getComponent().inject(this);
    }

    @NonNull
    @Override
    protected HelpPresenter getPresenter() {
        return helpPresenter;
    }

    @Override
    protected int getActivityLayoutId() {
        return R.layout.activity_help;
    }

    @OnClick(R.id.tv_help_info)
    public void helpInfoButtonClicked()
    {
        startActivity( new Intent(Intent.ACTION_DIAL, Uri.parse("tel:" + phoneNumber)));

    }
    @Override
    protected void configureViews() {
        super.configureViews();
        setToolbarHeader(getString(R.string.action_help));
        getPresenter().getRide();
    }

    @Override
    public void showDefaultLattisNumber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
        showPhoneNumberText(getPhoneNumberText(phoneNumber));
    }

    @Override
    public void showOperatorNumber(String phoneNumber) {
        this.phoneNumber=phoneNumber;
        showPhoneNumberText(getPhoneNumberText(phoneNumber));
    }

    void showPhoneNumberText(String helpText){
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
            helpInfo.setText(Html.fromHtml(helpText, Html.FROM_HTML_MODE_LEGACY));
        } else {
            helpInfo.setText(Html.fromHtml(helpText));
        }
    }


    String getPhoneNumberText(String phoneNumber){
        return getString(R.string.help_info, phoneNumber);
    }
    @Override
    protected void onInternetConnectionChanged(boolean isConnected) {

    }

}

File: velolabs_repos/lattis_android_v2/app/src/main/java/com/lattis/lattis/presentation/help/HelpActivity.kt
package com.lattis.lattis.presentation.help

import android.content.Intent
import android.net.Uri
import android.text.SpannableString
import android.text.style.UnderlineSpan
import android.view.View
import com.lattis.lattis.presentation.help.SliderImageHelper.imageSliderApplies
import com.lattis.lattis.presentation.help.SliderImageHelper.openSliderImage
import com.lattis.lattis.presentation.ui.base.activity.BaseActivity
import com.lattis.lattis.presentation.utils.FirebaseUtil
import com.lattis.lattis.presentation.webview.WebviewActivity
import io.lattis.lattis.R
import kotlinx.android.synthetic.main.activity_help.*
import kotlinx.android.synthetic.main.activity_loading.view.*
import kotlinx.android.synthetic.main.layout_image_slider_parent.view.*
import javax.inject.Inject

class HelpActivity : BaseActivity<HelpActivityPresenter, HelpActivityView>(),
    HelpActivityView {

    private val REQUEST_CODE_FAQ = 4394

    @Inject
    override lateinit var presenter: HelpActivityPresenter
    override val activityLayoutId = R.layout.activity_help
    override var view: HelpActivityView = this


    override fun configureViews() {
        super.configureViews()

        showLoadingForAddPromotion(getString(R.string.loading))

        if(imageSliderApplies()){
            btn_open_image_slider.visibility= View.VISIBLE
        }else{
            btn_open_image_slider.visibility = View.GONE
        }

        presenter.getHelpInfo()

        iv_close_in_help.setOnClickListener {
            finish()
        }



        ct_phone_number_label_help.setOnClickListener {
            startCalling()
        }

        ct_phone_number_value_help.setOnClickListener {
            startCalling()
        }


        ct_faq_label_help.setOnClickListener {
            openLink()
        }
        ct_faq_value_help.setOnClickListener {
            openLink()
        }


        btn_open_image_slider.setOnClickListener {
            openSliderImage(this, help_image_slider_parent)
        }

        help_image_slider_parent.ct_skip.setOnClickListener {
            help_image_slider_parent.visibility = View.GONE
        }

        FirebaseUtil.instance?.addCustomEvent(FirebaseUtil.HELP, FirebaseUtil.HELP)
    }

    fun startCalling(){
        startActivity(Intent(Intent.ACTION_DIAL, Uri.parse("tel:" + presenter.phoneNumber)))
        finish()
    }

    fun openLink(){
        WebviewActivity.launchForResult(
            this,
            REQUEST_CODE_FAQ, presenter.web_link
        )
    }

    override fun showPhoneNumber(phoneNumber: String) {
        ct_phone_number_label_help.visibility = View.VISIBLE
        ct_phone_number_value_help.visibility = View.VISIBLE

        ct_phone_number_value_help.text = phoneNumber
    }

    override fun showEmail(email: String) {
        ct_email_value_help.visibility = View.VISIBLE
        ct_email_label_help.visibility = View.VISIBLE

        ct_email_value_help.text = email
    }

    override fun showFaq(faq: String) {
        ct_faq_label_help.visibility = View.VISIBLE
        ct_faq_value_help.visibility = View.VISIBLE

        val content = SpannableString(faq)
        content.setSpan(UnderlineSpan(), 0, faq.length, 0)
        ct_faq_value_help.text = content
    }

    //// loading :start
    fun showLoadingForAddPromotion(message: String?) {
        help_activity_loading_view.visibility = (View.VISIBLE)
        help_activity_loading_view.ct_loading_title.text = (message)
    }

    override fun hideLoadingForAddPromotion() {
        help_activity_loading_view.visibility = (View.GONE)
    }



    override fun onInternetConnectionChanged(isConnected: Boolean) {

    }

}